devicetypes/smartthings/orvibo-gas-detector.src/Orvibo-Gas-detector.groovy

```diff
@@ -0,0 +1,123 @@
+ *  under the License.
+ *  Author : jinkang zhang / jk0218.zhang@samsung.com
+ *  Date : 2018-07-04
+ */
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition(name: "Orvibo Gas Detector", namespace: "smartthings", author: "SmartThings", runLocally: false, minHubCoreVersion: '000.017.0012', executeCommandsLocally: false, mnmn: "SmartThings", vid: "generic-smoke") {
+		capability "Smoke Detector"
+		capability "Configuration"
+		capability "Alarm"
+		capability "Health Check"
+		capability "Sensor"
+		capability "Refresh"
+		fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000,0003,0500,0009", outClusters: "0019",manufacturer: "Heiman",model:"d0e857bfd54f4a12816295db3945a421"
+	}
+	tiles {
+		standardTile("main", "device.smoke", width: 2, height: 2) {
+			state("clear", label:"Clear", icon:"st.alarm.smoke.clear", backgroundColor:"#ffffff")
+			state("detected", label:"Smoke!", icon:"st.alarm.smoke.smoke", backgroundColor:"#e86d13")
+		}
+		standardTile("alarm", "device.alarm", width: 2, height: 2) {
+			state "off", label:'off', action:'alarm.strobe', icon:"st.alarm.alarm.alarm", backgroundColor:"#ffffff"
+			state "strobe", label:'strobe!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+			state "siren", label:'siren!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+			state "both", label:'alarm!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		main "main"
+		details(["main","alarm","refresh"])
+	}
+}
+def installed(){
+	log.debug "installed"
+	return zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER,zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
+					zigbee.readAttribute(0x0009,0x0000)
+}
+def parse(String description) {
+	log.debug "description(): $description"
+	def map = zigbee.getEvent(description)
+	if(!map){
+		if (description?.startsWith('zone status')) {
+			map = parseIasMessage(description)
+		}
+	}
+	log.debug "Parse returned $map"
+	def result = map ? createEvent(map) : [:]
+	if (description?.startsWith('enroll request')) {
+		List cmds = zigbee.enrollResponse()
+		log.debug "enroll response: ${cmds}"
+		result = cmds?.collect { new physicalgraph.device.HubAction(it)}
+	}
+	return result
+}
+private Map parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	translateZoneStatus(zs)
+}
+private Map translateZoneStatus(ZoneStatus zs) {
+	// Some sensor models that use this DTH use alarm1 and some use alarm2 to signify motion
+	return (zs.isAlarm1Set() || zs.isAlarm2Set()) ? getDetectedResult('detected') : getDetectedResult('clear')
+}
+private Map getDetectedResult(value) {
```

Comment:
You are creating new method (translateZoneStatus) just to call return without any additional logic. Just call return here.
Or even better. Change getMotionResult(value) method. Check if value is true or false and create events based on that. Remeber to change stopMotion().